### DISCUSSION TITLE
CHANGE MY TABLE ID TO ANOTHER VALUE IN MY SELECT DEPENDENT

### DISCUSSION THREAD

**Discussion by vencomweb** [community/community#123355] [Programming Help] [answered]
### Body

Greetings friends I have a small and big problem at the same time I need in a dynamic select to save another value instead of the id in my database for example I select a redip and in the BD I save the name not the id... I would very much appreciate the support provided. I'll leave you the code and the database so you can find your way around
[SELECT DINAMY.docx](https://github.com/community/community/files/15281156/SELECT.DINAMY.docx)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by 0xf333** [comment]
@vencomweb 

Perhaps modifying the value attribute of the "redip" dropdown options from `$row['id_redip']` to `$row['redip']` should send the actual name of the selected "redip" instead of the ID when the form is submitted?

### e.g 
- Instead of approaching it as you're currently doing
```html
<div class="col-md">
  <div class="form-floating">
    <select name="redip" id="redip" class="form-select" aria-label="Default select example" required>
      <option value="">Seleccionar</option>
      <?php while($row = $estados->fetch_assoc()){ ?>
        <option value="<?php echo $row['id_redip']; ?>"><?php echo $row['redip']; ?></option>
      <?php } ?>
    </select>
    <label for="redip">Redip:</label>
  </div>
</div>
```

- You can perhaps try this instead and see
```html
<div class="col-md">
  <div class="form-floating">
    <select name="redip" id="redip" class="form-select" aria-label="Default select example" required>
      <option value="">Seleccionar</option>
      <?php while($row = $estados->fetch_assoc()){ ?>
        <option value="<?php echo $row['redip']; ?>"><?php echo $row['redip']; ?></option>
      <?php } ?>
    </select>
    <label for="redip">Redip:</label>
  </div>
</div>
```

**Comment by vencomweb** [comment, marked as answer]
Greetings my friend, I didn't find your support because when I made the change when selecting an element of the REDIP table in my DB it does not display the states assigned to that REDIP since the REDI ID calls the state ID. If you require more arguments, I would be happy to send you more information about the project and thank you always for the support

**Comment by vencomweb** [comment]
const cbxRedip = document.getElementById('redip')
cbxRedip.addEventListener('change', getEstados)

const cbxEstado = document.getElementById('estado')
cbxEstado.addEventListener('change', getMunicipios)

const cbxMunicipio = document.getElementById('municipio')
cbxMunicipio.addEventListener('change', getParroquias)

const cbxParroquia = document.getElementById('parroquia')
cbxParroquia.addEventListener('change', getCuadrantes)

const cbxCuadrante = document.getElementById('cuadrante')
cbxCuadrante.addEventListener('change', getOrganismos)

const cbxOrganismo = document.getElementById('organismo')

function fetchAndSetData(url, formData, targetElement) {
  return fetch(url, {
    method: "POST",
    body: formData,
    mode: 'cors',
  })
    .then(response => response.json())
    .then((data) => {
      targetElement.innerHTML = data
    })
    .catch((err) => console.log(err))
}

function getEstados() {
  let redip = cbxRedip.value
  let url = 'inc/getEstados.php'
  let formData = new FormData()
  formData.append('id_redip', redip)

  fetchAndSetData(url, formData, cbxEstado)
    .then(() => {
      cbxMunicipio.innerHTML = ''
      cbxMunicipio.innerHTML = "<option value=''>Seleccionar</option>"
    })
    .catch((err) => console.log(err))
}

function getMunicipios() {
  let estado = cbxEstado.value
  let url = 'inc/getMunicipios.php'
  let formData = new FormData()
  formData.append('id_estado', estado)

  fetchAndSetData(url, formData, cbxMunicipio)
    .then(() => {
      cbxParroquia.innerHTML = ''
      cbxParroquia.innerHTML = "<option value=''>Seleccionar</option>"
    })
    .catch((err) => console.log(err))
}

function getParroquias() {
  let municipio = cbxMunicipio.value
  let url = 'inc/getParroquias.php'
  let formData = new FormData();
  formData.append('id_municipio', municipio)

  fetchAndSetData(url, formData, cbxParroquia)
    .then(() => {
      cbxCuadrante.innerHTML = ''
      cbxCuadrante.innerHTML = "<option value=''>Seleccionar</option>"
    })
    .catch((err) => console.log(err))
}

function getCuadrantes() {
  let parroquia = cbxParroquia.value
  let url = 'inc/getCuadrantes.php'
  let formData = new FormData()
  formData.append('id_parroquia', parroquia)

  fetchAndSetData(url, formData, cbxCuadrante)
    .then(() => {
      cbxOrganismo.innerHTML = ''
      cbxOrganismo.innerHTML = "<option value=''>Seleccionar</option>"
    })
    .catch((err) => console.log(err))
}

function getOrganismos() {
  let cuadrante = cbxCuadrante.value
  let url = 'inc/getOrganismos.php'
  let formData = new FormData()
  formData.append('id_cuadrantePaz', cuadrante)

  fetchAndSetData(url, formData, cbxOrganismo)
    .catch(err => console.log(err)) 
}

**Comment by vencomweb** [comment]
These are the requests made to the dynamic selections

**Comment by vencomweb** [comment]
Any support from the community that can guide me

**Comment by 0xf333** [comment]
@vencomweb 
> Any support from the community that can guide me

Perhaps you could share the repo so people can easily locally reproduce the issue you're facing.
That way, community folks who want to help can actually see the problem in context and identify potential issues that may not be apparent from a small code snippet. In software development, "context is king"; meaning that understanding the broader context of a piece of code is kind of crucial for identifying and fixing bugs.
